share/shutter/resources/modules/Shutter/Screenshot/SelectorGnomeWayland.pm

```diff
@@ -0,0 +1,133 @@
+###################################################
+#
+#  Copyright (C) 2008-2013 Mario Kemper <mario.kemper@gmail.com>
+#
+#  This file is part of Shutter.
+#
+#  Shutter is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  Shutter is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with Shutter; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+###################################################
+
+#perl -x -S perltidy -l=0 -b "%f"
+
+package Shutter::Screenshot::SelectorGnomeWayland;
+
+#modules
+#--------------------------------------
+use utf8;
+use strict;
+use warnings;
+
+use Shutter::Screenshot::Main;
+use Shutter::Screenshot::History;
+
+use Data::Dumper;
+our @ISA = qw(Shutter::Screenshot::Main);
+
+#Glib
+use Glib qw/TRUE FALSE/;
+
+#--------------------------------------
+
+sub new {
+	my $class = shift;
+
+	#call constructor of super class (shutter_common, include_cursor, delay, notify_timeout)
+	my $self = $class->SUPER::new(shift, shift, shift, shift);
+
+	$self->{_hide_time}   = shift;    #a short timeout to give the server a chance to redraw the area that was obscured
+
+	$self->{_dpi_scale} = Gtk3::Window->new('toplevel')->get('scale-factor');
+
+	bless $self, $class;
+	return $self;
+}
+
+sub select_gnome_wayland {
+	my $self = shift;
+
+	#return value
+	my $output = 5;
+	
+	my $cmdcursor = undef;
+
+	if ($self->{_include_cursor}) {
+		$cmdcursor = "-p";
+	}
+	else {
+		$cmdcursor = "";
+	}
+	
+	my $fn = "/tmp/shutter-gnome-screenshot-tmp.png";
+	
+	system("gnome-screenshot", "-a", "-f", $fn, "-d", $self->{_delay}, $cmdcursor);
+	
+
+	
+	my $image = Gtk3::Image->new();
+	$image->set_from_file($fn);
+	$output = $image->get_pixbuf();
+
+	my $d = $self->{_sc}->get_gettext;
+
+	return $output;
+}
+
+
+sub redo_capture {
```

Comment:
Nope, this is just a remainder from the SelectorAdvance.pm module which I took as a base. I have some ideas, how redo could be possible (make a full screen screenshot with gnome-screenshot, then choose the proper area, but this would require to somehow get the information, what the proper area is, as now gnome-screenshot is responsible for doing the selection) but nothing working yet.